third_party/purge/ext.purge.php

```diff
@@ -100,7 +103,19 @@ public function send_purge_request()
 		
 		foreach ($urls as $url)
 		{
-			send_purge_request($url, $this->port);
+			//now loop through urls for this channel
+			$this->EE->db->select('*');
+			$this->EE->db->where('channel_id',(int) $meta['channel_id']);
+			$channelPatterns = $this->EE->db->get_where('purge_rules')->result_array();
+			
+			foreach($channelPatterns as $pattern)
+			{
+				$_pattern = str_replace('{url_title}',$meta['url_title'],$pattern['pattern']);
+				$_url = preg_replace('/\/$/','',$url).'/'.preg_replace('/^\//','',$_pattern); //handle trailing and beginning slashes
+				//echo $_url . "<br>";
+				send_purge_request($_url, $this->port);
+				unset($_pattern);
+			}
```

Comment:
I imagine this whole block can get pretty slow if you have a lot of patterns and multiple servers. Have you tried setting all of the patterns in a header and purging them all in one fell swoop per sever?